### DISCUSSION TITLE
How to update APISIX

### DISCUSSION THREAD

**Discussion by xxm404** [apache/apisix#6000] [Q&A]
How to update APISIX when we have some custom plugins? somebody can give me an opinion?

**Comment by xxm404** [comment]
The last time we updated APISIX from v2.1 to v2.5, I manually merged the code, the experience was very bad

**Reply by juzhiyuan** [reply]
AFAIK, usually, if there doesn't have breaking change, we need to first backup the ETCD, and then upgrade the apisix version.

For having the custom plugins case, do you have many conflicting changes?

also cc @leslie-tsang to have a look.
